From a6155a58d14e2819bb2a0b81b7344d39d073eae1 Mon Sep 17 00:00:00 2001
From: Wei Ouyang <oeway007@gmail.com>
Date: Mon, 12 Dec 2022 18:42:20 -0800
Subject: [PATCH] Fix zarr sync

This patch removes fasteners and disables zarr.sync which uses process and thread

Upstream PR:
https://github.com/zarr-developers/zarr-python/pull/1663

---
 pyproject.toml | 2 +-
 zarr/sync.py   | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 292bfddd..6d252925 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -14,7 +14,7 @@ requires-python = ">=3.8"
 dependencies = [
     'asciitree',
     'numpy>=1.20,!=1.21.0',
-    'fasteners',
+    'fasteners; sys_platform != "emscripten"',
     'numcodecs>=0.10.0',
 ]
 dynamic = [
diff --git a/zarr/sync.py b/zarr/sync.py
index 49684a51..0ce4baa7 100644
--- a/zarr/sync.py
+++ b/zarr/sync.py
@@ -2,8 +2,6 @@ import os
 from collections import defaultdict
 from threading import Lock
 
-import fasteners
-
 
 class ThreadSynchronizer:
     """Provides synchronization using thread locks."""
@@ -41,6 +39,7 @@ class ProcessSynchronizer:
         self.path = path
 
     def __getitem__(self, item):
+        import fasteners
         path = os.path.join(self.path, item)
         lock = fasteners.InterProcessLock(path)
         return lock
-- 
2.25.1

